addons/sourcemod/scripting/superzombiefortress.sp

```diff
@@ -657,6 +657,87 @@ public void OnEntityCreated(int iEntity, const char[] sClassname)
 	
 	if (StrEqual(sClassname, "tf_dropped_weapon") || StrEqual(sClassname, "item_powerup_rune"))	//Never allow dropped weapon and rune dropped from survivors
 		RemoveEntity(iEntity);
+		
+	if (StrEqual(sClassname, "prop_dynamic") && g_nRoundState == SZFRoundState_Active)
+	{
+		RequestFrame(Frame_SetSpawnedWeapon, iEntity);
+	}
+}
+
+public void Frame_SetSpawnedWeapon(int iWeapon)
```

Comment:
Can this whole function be merged with `Event_WeaponsRoundStart`? I really don't want to see another 70 lines of code being nearly identical to `Event_WeaponsRoundStart`.